packages/host/app/lib/matrix-classes/message-builder.ts

```diff
@@ -147,23 +147,53 @@ export default class MessageBuilder {
           r.event_id === this.builderContext.effectiveEventId)
       );
     }) as CommandResultEvent | undefined;
-    let status = (commandResultEvent?.content?.['m.relates_to']?.key ||
-      'ready') as CommandStatus;
-    let commandResultCardEventId =
-      commandResultEvent?.content?.msgtype ===
-      APP_BOXEL_COMMAND_RESULT_WITH_OUTPUT_MSGTYPE
-        ? commandResultEvent.content.data.cardEventId
-        : undefined;
-    let messageCommand = new MessageCommand(
+
+    return buildMessageCommand({
+      effectiveEventId: this.builderContext.effectiveEventId,
+      owner: getOwner(this)!,
       message,
-      command.id,
-      command.name,
-      command.arguments,
-      this.builderContext.effectiveEventId,
-      status,
-      commandResultCardEventId,
-      getOwner(this)!,
-    );
-    return messageCommand;
+      commandEvent: event,
+      commandResultEvent,
+    });
   }
 }
+
+export function buildMessageCommand({
+  effectiveEventId,
+  commandEvent,
+  commandResultEvent,
+  message,
+  owner,
+}: {
+  effectiveEventId: string;
+  commandEvent: CommandEvent;
+  commandResultEvent?: CommandResultEvent;
+  message: Message;
+  owner: Owner;
+}) {
+  let status = (commandResultEvent?.content['m.relates_to']?.key ||
+    'ready') as CommandStatus;
+  let commandResultCardEventId =
+    commandResultEvent?.content.msgtype ===
+    APP_BOXEL_COMMAND_RESULT_WITH_OUTPUT_MSGTYPE
+      ? commandResultEvent.content.data.cardEventId
+      : undefined;
+  let command = commandEvent.content.data.toolCall;
+  let messageCommand = new MessageCommand(
+    message,
+    command.id,
+    command.name,
+    command.arguments,
+    effectiveEventId,
+    status,
+    commandResultCardEventId,
+    owner,
+  );
+  return messageCommand;
+}
+
+export function formattedMessageForCommand(formattedBody: string) {
+  return `<p data-test-command-message class="command-message">${sanitizeHtml(
```

Comment:
@backspace  since we are not using triple curlies `{{{` this is not actually ever HTML. this is always only text. there is no need to sanitize this.